java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java

```diff
@@ -2300,7 +2293,11 @@ public static void cloneErrata(Long channelId, Collection<Long> errataToClone, b
                 }
                 else {
                     log.debug("Re-publishing clone");
-                    publish(clones.get(0), cids, user);
+                    Errata firstClone = clones.get(0);
```

Comment:
> call ErrataFactory.publish(), but that is a noop if the errata is
already published, which is always the case here

Is this true in general (outside CLM context)? This method is also used for general errata cloning via XMLRPC (`ChannelSoftwareHandler.mergeErrata`).

Reply:
It's not true in general, for example the Struts action PackagePushAction calls `publish` exclusively on unpublished erratas - and for those the behavior is definitely not no-op.